flaskr/api.py

```diff
@@ -16,48 +19,31 @@ def __init__(self):
 
     def get(self):
         # get all files in the ./static/uploads directory
-        files = [f for f in os.listdir(UPLOAD_DIR)]
+        files = [f for f in os.listdir(UPLOAD_DIR + '/dicoms')]
         return {
-                'dir' : UPLOAD_DIR, 
+                'pathRelative': DICOM_PATH_RELATIVE,
+                'pathAbsolute': DICOM_PATH_ABSOLUTE,
                 'count' : len(files),
                 'files' : files,
                 }
 
     def post(self):
-        self.parser.add_argument('startComputation', required=True, type=str)
-        self.parser.add_argument('CTFile', required=False, \
-                type=werkzeug.datastructures.FileStorage, location='files')
-        self.parser.add_arguemnt('DoseDataFile', required=False, \
-                type=werkzeug.datastructures.FileStorage, location='files')
-        self.parser.add_argument('RSFile', required=False, \
+        self.parser.add_argument('dicomArchive', required=True, \
                 type=werkzeug.datastructures.FileStorage, location='files')
 
         args = self.parser.parse_args()
 
-        if 'CTFile' in args:
-            save_file(args.get('CTFile'), 'CT.zip')
-        if 'DoseSumFile' in args:
-            save_file(args.get('DoseSumFile'), 'DoseSum.zip')
-        if 'RSFile' in args:
-            save_file(args.get('RSFile'), 'RS.zip')
-        
-        if args['startComputation']:
-            payload = {
-                    'CT_fname': 'CT.zip',
-                    'DoseSum_fname': 'DoseSum.zip',
-                    'RS_fname': 'RS.zip'
-                    }
-            response = requests.request('GET', GENIE_API+'load', headers={}, data=payload, files=[])
-            return {
-                    'successful': True,
-                    'computationStarted': True,
-                }
+        # save the .zip archive
+        archive = args.get('dicomArchive')
+        archive.save(os.path.join(UPLOAD_DIR, archive.filename))
```

Comment:
I'd suggest doing some basic checks: for example if the file is too big (let's say above 100MB) the request fails. Similarly an error is returned if uzipping fails. I'd also check if only DICOM files are in the archive (i.e. `archive.namelist` only contains files matching a regex like maybe `^[[:alnum:]\/\- \.\\]*\.dcm$`.

Reply:
I was hoping that the contents of the archive would be guaranteed on the front end, but it seems to be a good suggestion and I will definitely implement it here (it makes sense, as the back end should do this sort of computation)

Reply:
Another update: as it turns out, the zipfile module detfines a LargeZipFile exception, so I will use that instead of checking it manually